odoo/addons/test_access_rights/tests/test_ir_rules.py

```diff
@@ -124,3 +124,20 @@ def test_no_context_in_ir_rules(self):
         ObjCateg.clear_caches()
         records = SomeObj.with_context(only_media=True).search([('id', '=', self.id1)])
         self.assertTrue(records)
+
+    def test_domain_constrains(self):
+        """ An error should be raised if domain is not correct """
+
+        bad_domains = [
+            'A really bad domain!',
+            [(1, '!=', 1)],
+            [('non_existing_field', '=', 'value')],
+        ]
```

Comment:
Can you add a test with False as `domain_force` ? to cover the ` ... and rule.domain_force ...` condition?